### ISSUE TITLE
Feature: Warn about masked return value in `true "$(false)"`

### ISSUE THREAD

**Issue by bukzor** [koalaman/shellcheck#2144]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
set -o errexit

echo "$(echo FAILURE; exit 1)"

echo "errexit means there's been no failures before this line"
```

#### Here's what shellcheck currently says:

No issues detected!

#### Here's what I wanted or expected to see:

There should be a warning similar to [SC2155](https://github.com/koalaman/shellcheck/wiki/SC2155), and for the same reason -- the exit code is gobbled.

**Comment by bukzor** [comment]
If implemented, this check should warn about masked rreturn value for this example too:

```!sh
#!/bin/bash
set -e
X="$(echo "$(exit 1)")"
```

This is just a special case of the same problem -- the `echo` is successful due to ignoring the return code of its command-substituion argument.

**Comment by bukzor** [comment]
I found another:

```
#!/bin/sh
set -e
(
   false 
   true
 ) || echo "something went wrong"
 ```
 
This will not trigger the error message, unlike what the author intended.
